vending-app/src/main/java/org/microg/vending/billing/core/HttpClient.kt

```diff
@@ -1,140 +1,221 @@
 package org.microg.vending.billing.core
 
-import android.content.Context
-import android.net.Uri
-import com.android.volley.*
-import com.android.volley.toolbox.HttpHeaderParser
-import com.android.volley.toolbox.JsonObjectRequest
-import com.android.volley.toolbox.Volley
 import com.squareup.wire.Message
 import com.squareup.wire.ProtoAdapter
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.engine.okhttp.OkHttp
+import io.ktor.client.plugins.HttpTimeout
+import io.ktor.client.plugins.cache.HttpCache
+import io.ktor.client.plugins.timeout
+import io.ktor.client.request.forms.submitForm
+import io.ktor.client.request.get
+import io.ktor.client.request.headers
+import io.ktor.client.request.post
+import io.ktor.client.request.prepareGet
+import io.ktor.client.request.setBody
+import io.ktor.client.request.url
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.ParametersImpl
+import io.ktor.http.URLBuilder
+import io.ktor.http.Url
+import io.ktor.utils.io.ByteReadChannel
+import io.ktor.utils.io.pool.ByteArrayPool
 import org.json.JSONObject
 import org.microg.gms.utils.singleInstanceOf
-import kotlin.coroutines.resume
-import kotlin.coroutines.resumeWithException
-import kotlin.coroutines.suspendCoroutine
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.OutputStream
 
-private const val POST_TIMEOUT = 8000
+private const val POST_TIMEOUT = 8000L
 
-class HttpClient(context: Context) {
-    private val requestQueue = singleInstanceOf { Volley.newRequestQueue(context.applicationContext) }
+class HttpClient {
```

Comment:
May be renamed to DownloadClient if it makes more sens to you also ?

Reply:
@jonathanklee That doesn't make so much sense to me, as it not only downloads files, but also does other queries using both GET and POST.